speculation-rules.bs

```diff
@@ -649,6 +649,27 @@ The same mitigations available to other features also apply here. In particular,
 
 The possibility of leaking link URLs via this mechanism is additionally mitigated by the fact that prefetch and prerender to plaintext HTTP (other than to localhost) is not permitted, and so such an on-path attacker could not directly observe preloading request URLs, but would only have access to metadata and traffic analysis. This does not, however, replace standard XSS protections.
 
+<h3 id="security-inline-scripting">Inline scripting</h3>
+
+Inline script use is also restricted by the [[CSP]], but there is a case a developer doesn't want to permit general inline scripts, but want to permit inline speculation rules. So, we make the following change that extends [[CSP]] to support a new source keyword for inline speculation rules.
+
+In the [[CSP#framework-directive-source-list]], we change the following definition to have a new source, `inline-speculation-rules`.
+
+<pre dfn-type="grammar" link-type="grammar">
+  <dfn>keyword-source</dfn> = "<dfn>'self'</dfn>" / "<dfn>'unsafe-inline'</dfn>" / "<dfn>'unsafe-eval'</dfn>"
+    / "<dfn>'strict-dynamic'</dfn>" / "<dfn>'unsafe-hashes'</dfn>"
+    / "<dfn>'report-sample'</dfn>" / "<dfn>'unsafe-allow-redirects'</dfn>"
+    / "<dfn>'wasm-unsafe-eval'</dfn>" / "<dfn>'inline-speculation-rules'</dfn>"
+</pre>
+
+Also, we make a following change for a note in [[CSP#should-block-inline]].
+
+Note: The valid values for `type` are "`script`", "`script attribute`", "`script speculationrules`", "`style`", and "`style attribute`".
+
+Also we make the following change for [[CSP#directive-script-src]].
+
+  3.  Inline <{script}> blocks MUST pass through [[CSP#should-block-inline]]. Their behavior will be blocked unless every policy allows inline script, either implicitly by not specifying a `script-src` (or `default-src`) directive, or explicitly, by specifying "`unsafe-inline`", a <a grammar>nonce-source</a> or a <a grammar>hash-source</a> that matches the inline block, or specifying "`inline-speculation-rules`" on the type attribute being "`speculationrules`".
```

Comment:
I think you'd also want to adjust https://w3c.github.io/webappsec-csp/#match-element-to-source-list to actually perform the check you're suggesting here. The text you've edited isn't actually part of any algorithm, but expresses the intention behind the algorithms below. :)

Reply:
+1, this will need to patch either that or [allows all inline behavior](https://w3c.github.io/webappsec-csp/#source-list-allows-all-inline-behavior), not sure which.

Reply:
My last CL contains patches for both sections.